src/utils/api.js

```diff
@@ -31,6 +33,98 @@ export const formatActions = (actionsByDayObject) => {
   });
 };
 
+// Takes array of action objects and returns an object with {key: value} pairs of {hourOfDay: #-of-actions}
+export const getActionsByTimeOfDay = (actions) => {
+  const actionsByTimeOfDay = {
+    'Late night': 0,
+    'Early morning': 0,
+    Morning: 0,
+    Afternoon: 0,
+    Evening: 0,
+    Night: 0,
+  };
+  actions.forEach((action) => {
+    const actionHourOfDay = action.createdAt.slice(11, 13);
+    if (actionHourOfDay >= 0 && actionHourOfDay <= 4) {
+      actionsByTimeOfDay['Late night'] += 1;
+    } else if (actionHourOfDay > 4 && actionHourOfDay <= 8) {
+      actionsByTimeOfDay['Early morning'] += 1;
+    } else if (actionHourOfDay > 8 && actionHourOfDay <= 12) {
+      actionsByTimeOfDay.Morning += 1;
+    } else if (actionHourOfDay > 12 && actionHourOfDay <= 16) {
+      actionsByTimeOfDay.Afternoon += 1;
+    } else if (actionHourOfDay > 16 && actionHourOfDay <= 20) {
+      actionsByTimeOfDay.Evening += 1;
+    } else {
+      actionsByTimeOfDay.Night += 1;
+    }
+  });
+  return actionsByTimeOfDay;
+};
+
+// Takes object with {key: value} pairs of {timeOfDay: #-of-actions}
+// returns a date-sorted array in Recharts.js format
+export const formatActionsByTimeOfDay = (actionsByTimeOfDayObject) => {
+  const actionsByTimeOfDayArray = Object.entries(actionsByTimeOfDayObject);
+  return actionsByTimeOfDayArray.map((entry) => {
+    return {
+      timeOfDay: entry[0],
+      count: entry[1],
+    };
+  });
+};
+
+// Takes array of action objects and returns an object with {key: value} pairs of {verb: %-of-actions}
+export const getActionsByVerb = (actions) => {
+  const totalActions = actions.length;
+  const actionsByVerb = {};
+  actions.forEach((action) => {
+    if (!actionsByVerb[action.verb]) {
+      actionsByVerb[action.verb] = 1 / totalActions;
+    } else {
+      actionsByVerb[action.verb] += 1 / totalActions;
```

Comment:
Add parentheses or split into full statement to make it clear what you're expecting to have as a result.